src/backend/sgx/attestation.rs

```diff
@@ -0,0 +1,365 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// CREDITS
+// * https://github.com/fortanix/rust-sgx for examples of AESM requests.
+
+use crate::protobuf::aesm_proto::{
+    Request, Request_GetQuoteExRequest, Request_GetSupportedAttKeyIDNumRequest,
+    Request_GetSupportedAttKeyIDsRequest, Request_InitQuoteExRequest, Response,
+    Response_GetQuoteExResponse, Response_InitQuoteExResponse,
+};
+
+use std::io::{Error, ErrorKind, Read, Write};
+use std::mem::size_of;
+use std::os::unix::net::UnixStream;
+use std::slice::{from_raw_parts, from_raw_parts_mut};
+
+use protobuf::Message;
+use sallyport::syscall::{SGX_QUOTE_SIZE, SGX_TI_SIZE};
+
+const AESM_SOCKET: &str = "/var/run/aesmd/aesm.socket";
+const AESM_REQUEST_TIMEOUT: u32 = 1_000_000;
+const SGX_KEY_ID_SIZE: u32 = 256;
+
+// Specifies the protobuf Request type to communicate with AESMD.
+#[derive(Debug)]
+enum ReqType {
+    AkIdNum,
+    AkId,
+    TInfo,
+    KeySize,
+    Quote,
+}
+
+impl ReqType {
+    fn set_request(
+        &self,
+        report: Option<&[u8; 432]>,
+        akid: Option<Vec<u8>>,
+        size: Option<usize>,
+    ) -> Result<Request, Error> {
+        let mut req = Request::new();
+
+        match self {
+            ReqType::AkIdNum => {
+                // removed
```

Comment:
For a person unfamiliar with internals of this, the comment does not make it clearer - what exactly is removed? From where? It looks like there should be a `TODO` with an issue reference instead

Reply:
I added FIXME.